Python/Product/PythonTools/PythonTools/Extensions.cs

```diff
@@ -737,6 +739,24 @@ public static class Extensions {
             return (EnvDTE.DTE)provider.GetService(typeof(EnvDTE.DTE));
         }
 
+        internal static SVsShellDebugger GetShellDebugger(this IServiceProvider provider) {
+            return (SVsShellDebugger)provider.GetService(typeof(SVsShellDebugger));
+        }
+
+        internal static async System.Threading.Tasks.Task RefreshVariableViews(this IServiceProvider serviceProvider) {
+            EnvDTE.Debugger debugger = serviceProvider.GetDTE().Debugger;
+            foreach (AD7Engine engine in AD7Engine.GetEngines()) {
+                if (engine.Process != null && engine.Process.Id == debugger.CurrentProcess.ProcessID) {
```

Comment:
To be more specific about the way matching should be done, given a single `EnvDTE.Process2`:

- Check for `Process2.Transport.ID`. If it is our `PythonRemoteDebugPortSupplier`, then it's our remote process. Otherwise, it's either our local process, or a non-Python process.

  * If it's a remote process, find the engine with a matching process ID, for which `AD7Engine.PythonProcess` is `PythonRemoteProcess`, and its `Uri` matches `Process2.TransportQualifier`.

  * Otherwise, find the engine with a matching process ID, for which `AD7Engine.PythonProcess` is _not_ `PythonRemoteProcess`